src/aggregates/vector_selector.rs

```diff
@@ -56,6 +170,9 @@ fn vector_selector_transition_inner(
     unsafe {
         in_aggregate_context(fcinfo, || {
             let mut state = state.unwrap_or_else(|| {
+                if (bucket_width == 0) ^ (start_time == end_time) {
```

Comment:
**Issue:** Did you intend to use bitwise XOR here? It will do the wrong thing if both expressions are true (i.e. it will return false, even if one of the expressions is true). I think you probably meant to use logical OR `||` here, but perhaps I'm misunderstanding.

Reply:
From what I can tell, this is logical XOR: https://doc.rust-lang.org/reference/types/boolean.html#logical-xor